Removing kubectl get output e2e test

robscott opened the pull request:
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
This was primarily to prevent new API resources being added without proper kubectl integration. This test has been replaced in 1.18 and is now irrelevant (and flaky) here.

**Which issue(s) this PR fixes**:
Fixes #88335

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig cli
/priority critical-urgent
/assign @liggitt @tpepper @alejandrox1

liggitt commented:
Since the point of the test is to catch new resources, and https://github.com/kubernetes/kubernetes/pull/88966 will do that in 1.19, I would probably just delete this test from this branch.

robscott committed:
Removing kubectl get output e2e test

This was primarily to prevent new API resources being added without proper kubectl integration. This test has been replaced in 1.19 and is now irrelevant (and flaky) here.

robscott commented:
@liggitt Good call, I've updated this PR to do that.

liggitt commented on test/e2e/kubectl/kubectl.go:
```diff
@@ -419,110 +415,6 @@ var _ = SIGDescribe("Kubectl client", func() {
-						obj.SetName(obj.GetName() + randString)
-					}
-
-					createObjValidateOutputAndCleanup(client, obj, resource)
```
static check will complain about createObjValidateOutputAndCleanup and verbsContain being unused

robscott replied:
Thanks for catching that, I've removed those + the deleteObj helper function used by this test.

liggitt commented:
/lgtm
/approve

tpepper commented:
In order to bring test grid green ahead of tomorrow's planned patch release we're going to merge this, even though it's a tad close to release.  Risk is minor given nature of PR (test only on a fast running and frequent test) and ability to revert if for some reason the system goes amazingly emergent and merging this truly breaks something else.

justaugustus commented:
Slack discussion here for add'l context: https://kubernetes.slack.com/archives/C2C40FMNF/p1583975966431100
